src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java

```diff
@@ -18,17 +23,65 @@ public void onPlayerMove(PlayerMoveEvent event) {
         IridiumSkyblock.getInstance().getTeamManager().sendIslandBorder(event.getPlayer());
 
         user.getCurrentIsland().ifPresent(island -> {
-            if (event.getPlayer().getLocation().getY() < LocationUtils.getMinHeight(event.getPlayer().getWorld())) {
-                VoidEnhancementData voidEnhancementData = IridiumSkyblock.getInstance().getEnhancements().voidEnhancement.levels.get(IridiumSkyblock.getInstance().getTeamManager().getTeamEnhancement(island, "void").getLevel());
-                if (voidEnhancementData != null && voidEnhancementData.enabled) {
-                    if (IridiumSkyblock.getInstance().getTeamManager().teleport(event.getPlayer(), island.getHome(), island)) {
-                        event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().voidTeleport
-                                .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)
-                        ));
+            if (!(event.getPlayer().getLocation().getY() < LocationUtils.getMinHeight(event.getPlayer().getWorld()))) {
+                return;
+            }
+
+            VoidEnhancementData voidEnhancementData = IridiumSkyblock.getInstance()
+                    .getEnhancements().voidEnhancement.levels
+                    .get(IridiumSkyblock.getInstance().getTeamManager().getTeamEnhancement(island, "void").getLevel());
+
+            if (voidEnhancementData == null || !voidEnhancementData.enabled) {
+                return;
+            }
+
+            if (!IridiumSkyblock.getInstance().getTeamManager().teleport(event.getPlayer(), island.getHome(), island)) {
+                return;
+            }
+
+            event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().voidTeleport
+                    .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+
+            HashMap<ItemStack, Integer> lostItemsMap = new HashMap<>();
+            if (voidEnhancementData.itemLossChance > 0) {
+                for (ItemStack item : event.getPlayer().getInventory().getContents()) {
+                    if (item == null) {
+                        continue;
                     }
+
+                    ItemStack originalItem = item.clone();
+
+                    for (int i = 0; i < item.getAmount(); i++) {
+                        if (Math.random() * 100 > voidEnhancementData.itemLossChance) {
+                            continue;
+                        }
+
+                        lostItemsMap.compute(originalItem, (key, value) -> value == null ? 1 : value + 1);
+                    }
+
+                    item.setAmount(item.getAmount() - lostItemsMap.getOrDefault(originalItem, 0));
```

Comment:
Sorry, it took me a while to figure out what this was doing...

Would this make more sence?

```suggestion
                    ArrayList<ItemStack> lostItems = new ArrayList<>();
                    ItemStack originalItem = item.clone();
                    int lostAmount = 0;

                    for (int i = 0; i < item.getAmount(); i++) {
                        if (Math.random() * 100 <= voidEnhancementData.itemLossChance) {
                            lostAmount++;
                        }
                    }
                    if(lostAmount == 0) continue;
                    origionalItem.setAmount(lostAmount);
                    lostItemsMap.add(origionalItem);

                    int newAmount = originalItem.getAmount() - lostAmount;
                    item.setAmount(newAmount);
```

Reply:
implemented this in a different commit, but I did change it up a bit, so please take a looksie